File: server/src/main/scala/kpn/api/common/planner/PlanLegDetail.scala
package kpn.api.common.planner

import kpn.api.common.common.ToStringBuilder

case class PlanLegDetail(
  source: LegEnd,
  sink: LegEnd,
  routes: Seq[PlanRoute]
) {

  def meters: Long = routes.map(_.meters).sum

  override def toString: String = ToStringBuilder(this.getClass.getSimpleName).
    field("routes", routes).
    build

}

File: backend/api/src/main/java/kpn/api/common/planner/PlanLegDetail.java
package kpn.api.common.planner;

import kpn.api.common.planner.LegEnd;
import kpn.api.common.planner.PlanRoute;

import com.google.common.collect.ImmutableList;

public record PlanLegDetail(
  LegEnd source,
  LegEnd sink,
  ImmutableList<PlanRoute> routes
) {
}

/* TODO migrate

  def meters: Long = routes.map(_.meters).sum

*/
